File: Python/eigen_calculation.py
# eigen_calculation.py
# Calculates the eigenvalues and eigenvectors of a matrix

import numpy as np

def calculate_eigen(matrix):
    """
    This function calculates the eigenvalues and eigenvectors of a given square matrix.
    
    Parameters:
    matrix (list of lists): A square matrix for which eigenvalues and eigenvectors are calculated.
    
    Returns:
    tuple: A tuple containing the eigenvalues and eigenvectors as numpy arrays.
    """
    # Convert input matrix to numpy array
    matrix = np.array(matrix)
    
    # Check if matrix is square
    if matrix.shape[0] != matrix.shape[1]:
        raise ValueError("The matrix must be square to calculate eigenvalues and eigenvectors.")
    
    # Calculate eigenvalues and eigenvectors
    eigenvalues, eigenvectors = np.linalg.eig(matrix)
    return eigenvalues, eigenvectors

File: R/eigen_calculation.R
# eigen_calculation.R
# Calculates eigenvalues and eigenvectors of a matrix

# Function to compute eigenvalues and eigenvectors
calculate_eigen <- function(matrix) {
  # Check if the matrix is square
  if (nrow(matrix) != ncol(matrix)) {
    stop("Matrix must be square to calculate eigenvalues and eigenvectors.")
  }
  
  # Calculate eigenvalues and eigenvectors
  eigen_results <- eigen(matrix)
  
  # Extract eigenvalues and eigenvectors
  eigenvalues <- eigen_results$values
  eigenvectors <- eigen_results$vectors
  
  return(list("eigenvalues" = eigenvalues, "eigenvectors" = eigenvectors))
}
